nucleus/deployment/admin/src/main/java/fish/payara/deployment/admin/PayaraTransformer.java

```diff
@@ -0,0 +1,112 @@
+ * https://github.com/payara/Payara/blob/master/LICENSE.txt
+ * See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * The Payara Foundation designates this particular file as subject to the "Classpath"
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.deployment.admin;
+
+import com.sun.enterprise.util.LocalStringManagerImpl;
+import java.io.File;
+import java.io.IOException;
+import static org.eclipse.transformer.Transformer.SUCCESS_RC;
+import org.eclipse.transformer.payara.JakartaNamespaceTransformer;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.hk2.classmodel.reflect.Type;
+import org.glassfish.hk2.classmodel.reflect.Types;
+
+/**
+ * Transforms jakarta.* namespace to javax.* and vice-versa.
+ * 
+ * @author gaurav.gupta@payara.fish
+ */
+public class PayaraTransformer {
+
+    public static final String TRANSFORM_NAMESPACE = "fish.payara.deployment.transform.namespace";
+
+    private static final LocalStringManagerImpl localStrings = new LocalStringManagerImpl(PayaraTransformer.class);
+
+    private static final String[] commonJavaxClasses = {
+        "javax.inject.Inject",
+        "javax.servlet.http.HttpServlet",
+        "javax.ws.rs.core.Application",
+        "javax.persistence.Entity"
+    };
+    private static final String[] commonJakartaClasses = {
+        "jakarta.inject.Inject",
+        "jakarta.servlet.http.HttpServlet",
+        "jakarta.ws.rs.core.Application",
+        "jakarta.persistence.Entity"
+    };
+    private static final String JAKARTA_PACKAGE_PREFIX = "jakarta.";
+
+    public static boolean isJakartaEEApplication(Types types) {
+
+        // Quick check for the most common Javax/Jakarta APIs
```

Comment:
Don't we already have a sniffer to do this sort of thing? (Note: I recognise this check may take place before sniffers have gotten involved, just checking that it's been a consideration)

Reply:
Hi @MattGill98, 

Yes, you are right, Jakarta Application identification operation performed during `preAuthorization` phase where sniffer details are not available.